### PR TITLE
Acr API version upgrade, support network ruleset, new cmdlets for usage/import image/login

### PR DESCRIPTION
**Pull request by VeryEarly**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/ContainerRegistry/ContainerRegistry/Commands/ConnectAzureContainerRegistry.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            {
+                //try to get registry from current subscription
+                IList<PSContainerRegistry> registries = this.RegistryClient.ListAllRegistries(null);
+                PSContainerRegistry registry = null;
+                if (registries != null || registries.Count != 0)
+                {
+                    registry = registries.FirstOrDefault(x => x.Name.Equals(Name));
+                }
+
+                if (registry != null && registry.SkuName.Equals("Classic"))
```
Comment (yugangw-msft) [inline src/ContainerRegistry/ContainerRegistry/Commands/ConnectAzureContainerRegistry.cs:67]: Like I said, you can remove all those logics on `Classic`

File: src/ContainerRegistry/ContainerRegistry/Commands/NewAzureContainerRegistry.cs [hunk 2]
```diff
@@ -96,7 +101,8 @@ public override void ExecuteCmdlet()
                         Sku = new Microsoft.Azure.Management.ContainerRegistry.Models.Sku(Sku),
                         AdminUserEnabled = EnableAdminUser,
                         Tags = tags,
-                        Location = Location
+                        Location = Location,
+                        NetworkRuleSet = NetworkRuleSet?.GetNetworkRuleSet()
```
Comment (yugangw-msft) [inline src/ContainerRegistry/ContainerRegistry/Commands/NewAzureContainerRegistry.cs:105]: I suggest we don't expose it from `create`, as we don't plan to GA this; and making it possible from `update` cmdlet is enough.

File: src/ContainerRegistry/ContainerRegistry/Commands/UpdateAzureContainerRegistry.cs [hunk 3]
```diff
@@ -104,7 +109,7 @@ public override void ExecuteCmdlet()
 
             if (ShouldProcess(Name, "Update Container Registry"))
             {
-                var registry = RegistryClient.UpdateRegistry(ResourceGroupName, Name, adminUserEnabled, Sku, storageAccountId, tags);
+                var registry = RegistryClient.UpdateRegistry(ResourceGroupName, Name, adminUserEnabled, Sku, storageAccountId, NetworkRuleSet, tags);
                 WriteObject(new PSContainerRegistry(registry));
             }
         }
```
Comment (yugangw-msft) [inline src/ContainerRegistry/ContainerRegistry/Commands/UpdateAzureContainerRegistry.cs:115]: For phase two, I suggest expose:

- AllowPublicNetworkAccess
-  Enable/Disable data endpoint

**Review by yugangw-msft — APPROVED** [review]

**Comment by isra-fel** [comment]
LGTM

**Review by isra-fel — APPROVED** [review]

**Pull request merged**